.github/workflows/ci.yml

```diff
@@ -15,25 +15,44 @@ jobs:
     timeout-minutes: 40
     strategy:
       matrix:
-        platform: ['ubuntu-20.04', 'macos-11']
+        platform: ['ubuntu-22.04', 'macos-11']
         python: ['3.7', '3.8', '3.9', '3.10', '3.11']
+        redis-version: ['7.0']
       fail-fast: false
     defaults:
       run:
         shell: bash -l -eo pipefail {0}
 
     steps:
-    - uses: actions/checkout@v3
+    - name: checkout
+      uses: actions/checkout@v3
       # Number of commits to fetch. 0 indicates all history for all branches and tags.
       with:
         fetch-depth: ''
 
+    - name: clone redis
+      uses: actions/checkout@v3
+      # Number of commits to fetch. 0 indicates all history for all branches and tags.
+      with:
+        fetch-depth: ''
+        repository: 'redis/redis'
+        ref: ${{matrix.redis-version}}
+        path: redis
+
     - name: Setup Python
       uses: actions/setup-python@v4
       with:
         python-version: ${{ matrix.python }}
         architecture: x64
 
+    - name: Setup Poetry
+      uses: snok/install-poetry@v1
+      with:
+          version: latest
+          virtualenvs-in-project: true
+          virtualenvs-create: true
+          installer-parallel: true
+
     - name: Cache pip
       uses: actions/cache@v3
       with:
```

Comment:
If we're already here, poetry maintains a separate cache from pip in `~/.cache/pypoetry`. Add?